Path to Mac SDK missing when building for osx with Cocoapods

Attempting to build extension-firebase for osx (adding a Podfile, allowing firebase_ios.mm to also build for DM_PLATFORM_OSX) fails with errors such as:

```
In file included from CocoaPodsService/Pods/PromisesObjC/Sources/FBLPromises/FBLPromise+Delay.m:17:
In file included from CocoaPodsService/Pods/PromisesObjC/Sources/FBLPromises/include/FBLPromise+Delay.h:17:
In file included from CocoaPodsService/Pods/PromisesObjC/Sources/FBLPromises/include/FBLPromise.h:17:
CocoaPodsService/Pods/PromisesObjC/Sources/FBLPromises/include/FBLPromiseError.h:17:9: fatal error: 'Foundation/Foundation.h' file not found
#import <Foundation/Foundation.h>
```

Examining the logs for the failed build, a typical compiler invocation looks like this:
<details>
<summary>Log from osx extension-firebase build</summary>
clang++ -c  -DDLIB_LOG_DOMAIN="PROMISESOBJC" -DDDF_EXPOSE_DESCRIPTORS -DDM_PLATFORM_MACOS -DDM_PLATFORM_OSX -DGL_DO_NOT_WARN_IF_MULTI_GL_VERSION_HEADERS_INCLUDED -DDM_DEBUG  -fobjc-arc --language=objective-c -fmodules -fmodule-name=FBLPromises -fmodule-map-file=/var/folders/ny/05wk595149v2zs2zz5jtwct40000gn/T/job8963115461759210323/CocoaPodsService/generated/PromisesObjC/module.modulemap  -Iupload/ -Iupload/firebase/include/ -ICocoaPodsService/Pods/PromisesObjC/Sources/FBLPromises/include/ -ICocoaPodsService/Pods/PromisesObjC/Sources/FBLPromises/ -ICocoaPodsService/Pods/PromisesObjC/ -ICocoaPodsService/generated/PromisesObjC/ -ICocoaPodsService/Pods/GoogleUtilities/ -ICocoaPodsService/generated/Privacy/ -ICocoaPodsService/Pods/GoogleUtilities/GoogleUtilities/Environment/Public/GoogleUtilities/ -ICocoaPodsService/Pods/GoogleUtilities/GoogleUtilities/Environment/Public/ -ICocoaPodsService/Pods/GoogleUtilities/third_party/IsAppEncrypted/Public/ -ICocoaPodsService/Pods/GoogleUtilities/third_party/IsAppEncrypted/ -ICocoaPodsService/Pods/GoogleUtilities/ -ICocoaPodsService/generated/Environment/ -ICocoaPodsService/Pods/GoogleUtilities/GoogleUtilities/Logger/Public/GoogleUtilities/ -ICocoaPodsService/Pods/GoogleUtilities/GoogleUtilities/Logger/Public/ -ICocoaPodsService/Pods/GoogleUtilities/ -ICocoaPodsService/generated/Logger/ -ICocoaPodsService/Pods/GoogleUtilities/GoogleUtilities/UserDefaults/Public/GoogleUtilities/ -ICocoaPodsService/Pods/GoogleUtilities/GoogleUtilities/UserDefaults/Public/ -ICocoaPodsService/Pods/GoogleUtilities/ -ICocoaPodsService/generated/UserDefaults/ -ICocoaPodsService/Pods/GoogleUtilities/GoogleUtilities/NSData+zlib/Public/GoogleUtilities/ -ICocoaPodsService/Pods/GoogleUtilities/GoogleUtilities/NSData+zlib/Public/ -ICocoaPodsService/Pods/GoogleUtilities/ -ICocoaPodsService/generated/NSData+zlib/ -ICocoaPodsService/Pods/FirebaseCoreInternal/ -ICocoaPodsService/generated/FirebaseCoreInternal/ -ICocoaPodsService/Pods/FirebaseCore/FirebaseCore/Sources/Public/FirebaseCore/ -ICocoaPodsService/Pods/FirebaseCore/FirebaseCore/Sources/Public/ -ICocoaPodsService/Pods/FirebaseCore/FirebaseCore/Sources/ -ICocoaPodsService/Pods/FirebaseCore/FirebaseCore/ -ICocoaPodsService/Pods/FirebaseCore/FirebaseCore/Extension/ -ICocoaPodsService/Pods/FirebaseCore/ -ICocoaPodsService/generated/FirebaseCore/ -ICocoaPodsService/Pods/FirebaseInstallations/FirebaseInstallations/Source/Library/InstallationsIDController/ -ICocoaPodsService/Pods/FirebaseInstallations/FirebaseInstallations/Source/Library/ -ICocoaPodsService/Pods/FirebaseInstallations/FirebaseInstallations/Source/Library/Public/FirebaseInstallations/ -ICocoaPodsService/Pods/FirebaseInstallations/FirebaseInstallations/Source/Library/Public/ -ICocoaPodsService/Pods/FirebaseInstallations/FirebaseInstallations/Source/Library/Private/ -ICocoaPodsService/Pods/FirebaseInstallations/FirebaseInstallations/Source/Library/InstallationsAPI/ -ICocoaPodsService/Pods/FirebaseInstallations/FirebaseInstallations/Source/Library/Errors/ -ICocoaPodsService/Pods/FirebaseInstallations/FirebaseInstallations/Source/Library/IIDMigration/ -ICocoaPodsService/Pods/FirebaseInstallations/FirebaseInstallations/Source/Library/InstallationsStore/ -ICocoaPodsService/Pods/FirebaseInstallations/FirebaseInstallations/Source/ -ICocoaPodsService/Pods/FirebaseInstallations/FirebaseCore/Extension/ -ICocoaPodsService/Pods/FirebaseInstallations/FirebaseCore/ -ICocoaPodsService/Pods/FirebaseInstallations/ -ICocoaPodsService/generated/FirebaseInstallations/   -I/usr/local/extender/sdk/37a4a8548850a7243055bb820e45580cfdae5f32/defoldsdk//include -I/usr/local/extender/sdk/37a4a8548850a7243055bb820e45580cfdae5f32/defoldsdk//sdk/include -I/usr/local/extender/sdk/37a4a8548850a7243055bb820e45580cfdae5f32/defoldsdk//ext/include  CocoaPodsService/Pods/PromisesObjC/Sources/FBLPromises/FBLPromise+Do.m -obuild/FBLPromise+Do.m_7.o
</details>

whereas the equivalent for a successful iOS build is:
<details>
<summary>Log from ios extension-firebase build</summary>
clang++ -c -arch arm64 -target arm-apple-darwin19 -m64 -O2 -g -stdlib=libc++ -miphoneos-version-min=11.0 -isysroot /usr/local/extender-stage/platformsdk/iPhoneOS16.2.sdk -nostdinc++  -DDLIB_LOG_DOMAIN="PROMISESOBJC" -DDDF_EXPOSE_DESCRIPTORS -DDM_PLATFORM_IOS -DDM_DEBUG  -fobjc-arc --language=objective-c -fmodules -fmodule-name=FBLPromises -fmodule-map-file=/var/folders/2b/mpxyl1m54jvg2rcs2cnbsqxc0000gn/T/job9297218435465495137/CocoaPodsService/generated/PromisesObjC/module.modulemap  -Iupload/ -Iupload/firebase/include/ -ICocoaPodsService/Pods/PromisesObjC/Sources/FBLPromises/include/ -ICocoaPodsService/Pods/PromisesObjC/Sources/FBLPromises/ -ICocoaPodsService/Pods/PromisesObjC/ -ICocoaPodsService/generated/PromisesObjC/ -ICocoaPodsService/Pods/GoogleUtilities/ -ICocoaPodsService/generated/Privacy/ -ICocoaPodsService/Pods/GoogleUtilities/GoogleUtilities/Environment/Public/GoogleUtilities/ -ICocoaPodsService/Pods/GoogleUtilities/GoogleUtilities/Environment/Public/ -ICocoaPodsService/Pods/GoogleUtilities/third_party/IsAppEncrypted/Public/ -ICocoaPodsService/Pods/GoogleUtilities/third_party/IsAppEncrypted/ -ICocoaPodsService/Pods/GoogleUtilities/ -ICocoaPodsService/generated/Environment/ -ICocoaPodsService/Pods/GoogleUtilities/GoogleUtilities/Logger/Public/GoogleUtilities/ -ICocoaPodsService/Pods/GoogleUtilities/GoogleUtilities/Logger/Public/ -ICocoaPodsService/Pods/GoogleUtilities/ -ICocoaPodsService/generated/Logger/ -ICocoaPodsService/Pods/GoogleUtilities/GoogleUtilities/UserDefaults/Public/GoogleUtilities/ -ICocoaPodsService/Pods/GoogleUtilities/GoogleUtilities/UserDefaults/Public/ -ICocoaPodsService/Pods/GoogleUtilities/ -ICocoaPodsService/generated/UserDefaults/ -ICocoaPodsService/Pods/GoogleUtilities/GoogleUtilities/NSData+zlib/Public/GoogleUtilities/ -ICocoaPodsService/Pods/GoogleUtilities/GoogleUtilities/NSData+zlib/Public/ -ICocoaPodsService/Pods/GoogleUtilities/ -ICocoaPodsService/generated/NSData+zlib/ -ICocoaPodsService/Pods/FirebaseCoreInternal/ -ICocoaPodsService/generated/FirebaseCoreInternal/ -ICocoaPodsService/Pods/FirebaseCore/FirebaseCore/Sources/Public/FirebaseCore/ -ICocoaPodsService/Pods/FirebaseCore/FirebaseCore/Sources/Public/ -ICocoaPodsService/Pods/FirebaseCore/FirebaseCore/Sources/ -ICocoaPodsService/Pods/FirebaseCore/FirebaseCore/ -ICocoaPodsService/Pods/FirebaseCore/FirebaseCore/Extension/ -ICocoaPodsService/Pods/FirebaseCore/ -ICocoaPodsService/generated/FirebaseCore/ -ICocoaPodsService/Pods/FirebaseInstallations/FirebaseInstallations/Source/Library/InstallationsIDController/ -ICocoaPodsService/Pods/FirebaseInstallations/FirebaseInstallations/Source/Library/ -ICocoaPodsService/Pods/FirebaseInstallations/FirebaseInstallations/Source/Library/Public/FirebaseInstallations/ -ICocoaPodsService/Pods/FirebaseInstallations/FirebaseInstallations/Source/Library/Public/ -ICocoaPodsService/Pods/FirebaseInstallations/FirebaseInstallations/Source/Library/Private/ -ICocoaPodsService/Pods/FirebaseInstallations/FirebaseInstallations/Source/Library/InstallationsAPI/ -ICocoaPodsService/Pods/FirebaseInstallations/FirebaseInstallations/Source/Library/Errors/ -ICocoaPodsService/Pods/FirebaseInstallations/FirebaseInstallations/Source/Library/IIDMigration/ -ICocoaPodsService/Pods/FirebaseInstallations/FirebaseInstallations/Source/Library/InstallationsStore/ -ICocoaPodsService/Pods/FirebaseInstallations/FirebaseInstallations/Source/ -ICocoaPodsService/Pods/FirebaseInstallations/FirebaseCore/Extension/ -ICocoaPodsService/Pods/FirebaseInstallations/FirebaseCore/ -ICocoaPodsService/Pods/FirebaseInstallations/ -ICocoaPodsService/generated/FirebaseInstallations/ -ICocoaPodsService/frameworks/headers/arm64-ios/  -F/var/folders/2b/mpxyl1m54jvg2rcs2cnbsqxc0000gn/T/job9297218435465495137/CocoaPodsService/frameworks/lib/arm64-ios  -I/usr/local/extender-stage/sdk/37a4a8548850a7243055bb820e45580cfdae5f32/defoldsdk//include -I/usr/local/extender-stage/sdk/37a4a8548850a7243055bb820e45580cfdae5f32/defoldsdk//sdk/include -I/usr/local/extender-stage/sdk/37a4a8548850a7243055bb820e45580cfdae5f32/defoldsdk//ext/include   CocoaPodsService/Pods/PromisesObjC/Sources/FBLPromises/FBLPromise+Do.m -obuild/FBLPromise+Do.m_7.o
</details>

I'm suspicious that a path to the Mac SDK is not specified with -isysroot, which would normally be the case, i.e. this is an empty objective-c file being built for osx in a test project:
<details>
<summary>Log from osx non-podfile objective-c file compile</summary>
clang++ -c -isystem /usr/local/extender/platformsdk/MacOSX13.1.sdk/usr/include/c++/v1  -DDLIB_LOG_DOMAIN="DEFOLDIMGUI" -DDDF_EXPOSE_DESCRIPTORS -DDM_PLATFORM_MACOS -DDM_PLATFORM_OSX -DGL_DO_NOT_WARN_IF_MULTI_GL_VERSION_HEADERS_INCLUDED  -O2 -g -stdlib=libc++ -mmacosx-version-min=10.13 -isysroot /usr/local/extender/platformsdk/MacOSX13.1.sdk -nostdinc++ -fno-exceptions -fvisibility=hidden -Werror=format -arch arm64 -target arm64-apple-darwin19 -m64  -Iupload/imgui/include/ -Ibuild/imgui/ -Iupload/   -I/usr/local/extender/sdk/37a4a8548850a7243055bb820e45580cfdae5f32/defoldsdk//include -I/usr/local/extender/sdk/37a4a8548850a7243055bb820e45580cfdae5f32/defoldsdk//sdk/include -I/usr/local/extender/sdk/37a4a8548850a7243055bb820e45580cfdae5f32/defoldsdk//ext/include  upload/imgui/src/empty_file.m -obuild/empty_file.m_0.o
</details>

NB this is when running a local standalone server, on [this commit](https://github.com/defold/extender/commit/06b486ad28b37e38376ae7504c8835ed4e0f5f61)

--- a/server/src/main/java/com/defold/extender/services/cocoapods/PodUtils.java
+++ b/server/src/main/java/com/defold/extender/services/cocoapods/PodUtils.java
@@ -103,7 +103,9 @@ public class PodUtils {
         switch(extenderTargetPlatform) {
             case "arm64-ios": return "arm64-apple-ios";
             case "x86_64-ios": return "x86_64-apple-ios-simulator";
+            case "arm64-osx":
             case "arm64-macos": return "arm64-apple-macos";
+            case "x86_64-osx":
             case "x86_64-macos": return "x86_64-apple-macos";
             default:
                 throw new ExtenderException(String.format("Invalid platform input for swift module name %s", extenderTargetPlatform));

--- a/server/src/test/java/com/defold/extender/ExtenderTest.java
+++ b/server/src/test/java/com/defold/extender/ExtenderTest.java
@@ -53,6 +53,19 @@ public class ExtenderTest {
         assertEquals(Arrays.asList(expected), l);
     }
 
+    static PlatformConfig mergePlatformConfig(Configuration config, String platform) throws ExtenderException
+    {
+        PlatformConfig platformConfig = new PlatformConfig();
+        platformConfig.context = new HashMap<>(config.context);
+        for (String platformAlt : ExtenderUtil.getPlatformAlternatives(platform)) {
+            PlatformConfig platformConfigAlt = config.platforms.get(platformAlt);
+            if (platformConfigAlt != null) {
+                ExtenderUtil.mergeObjects(platformConfig, platformConfigAlt);
+            }
+        }
+        return platformConfig;
+    }
+
     @Test
     public void testExtender(@TempDir File jobDir) throws IOException, InterruptedException, ExtenderException {
         File uploadDir = new File(jobDir, "upload");
@@ -387,6 +400,57 @@ public class ExtenderTest {
 
         uploadDir.delete();
         assertTrue(true);
+    }
+
+    @Test
+    public void testOsxCompileCommandCarriesSdkFlags() throws IOException, ExtenderException {
+        File root = new File("test-data");
+        File sdk = new File(root, "sdk/a/defoldsdk");
+        Configuration config = Extender.loadYaml(root, new File(sdk, "extender/build.yml"), Configuration.class);
+
+        PlatformConfig platformConfig = mergePlatformConfig(config, "arm64-osx");
+
+        assertTrue(platformConfig.compileCmd.contains("{{#clangArch}}-arch {{{clangArch}}} {{/clangArch}}"));
+        assertTrue(platformConfig.compileCmd.contains("{{#clangTarget}}-target {{{clangTarget}}} {{/clangTarget}}"));
+        assertTrue(platformConfig.compileCmd.contains("{{#clangArch}}-m64 {{/clangArch}}"));
+        assertTrue(platformConfig.compileCmd.contains("-isysroot {{env.SYSROOT}}"));
+        assertEquals("arm64", platformConfig.context.get("clangArch"));
+        assertEquals("arm64-apple-darwin19", platformConfig.context.get("clangTarget"));
+
+        List<String> flags = (List<String>)platformConfig.context.get("flags");
+        assertFalse(flags.contains("-arch"));
+        assertFalse(flags.contains("-target"));
+        assertFalse(flags.contains("-isysroot"));
+        assertFalse(flags.contains("{{env.SYSROOT}}"));
+
+        Map<String, String> env = new HashMap<>();
+        env.put("SYSROOT", "/opt/platformsdk/MacOSX.sdk");
+        env.put("MACOS_VERSION_MIN", "10.15");
+
+        Map<String, Object> renderContext = new HashMap<>(platformConfig.context);
+        renderContext.put("env", env);
+        renderContext.put("ext", new HashMap<String, Object>());
+        renderContext.put("extension_name_upper", "TEST");
+        renderContext.put("src", "source.mm");
+        renderContext.put("tgt", "source.o");
+        String renderedCompileCmd = new TemplateExecutor().execute(platformConfig.compileCmd, renderContext);
+        assertTrue(renderedCompileCmd.contains("-arch arm64"));
+        assertTrue(renderedCompileCmd.contains("-target arm64-apple-darwin19"));
+        assertTrue(renderedCompileCmd.contains("-m64"));
+        assertTrue(renderedCompileCmd.contains("-isysroot /opt/platformsdk/MacOSX.sdk"));
+
+        PlatformConfig legacyPlatformConfig = mergePlatformConfig(config, "x86-osx");
+        Map<String, Object> legacyRenderContext = new HashMap<>(legacyPlatformConfig.context);
+        legacyRenderContext.put("env", env);
+        legacyRenderContext.put("ext", new HashMap<String, Object>());
+        legacyRenderContext.put("extension_name_upper", "TEST");
+        legacyRenderContext.put("src", "source.mm");
+        legacyRenderContext.put("tgt", "source.o");
+        String legacyRenderedCompileCmd = new TemplateExecutor().execute(legacyPlatformConfig.compileCmd, legacyRenderContext);
+        assertFalse(legacyRenderedCompileCmd.contains("-arch "));
+        assertFalse(legacyRenderedCompileCmd.contains("-target "));
+        assertFalse(legacyRenderedCompileCmd.contains("-m64 "));
+        assertTrue(legacyRenderedCompileCmd.contains("-isysroot /opt/platformsdk/MacOSX.sdk"));
     }
 
     @Test

--- a/server/src/test/java/com/defold/extender/ExtenderTest.java
+++ b/server/src/test/java/com/defold/extender/ExtenderTest.java
@@ -416,12 +416,16 @@ public class ExtenderTest {
         assertTrue(platformConfig.compileCmd.contains("-isysroot {{env.SYSROOT}}"));
         assertEquals("arm64", platformConfig.context.get("clangArch"));
         assertEquals("arm64-apple-darwin19", platformConfig.context.get("clangTarget"));
+        List<String> swiftFlags = (List<String>)platformConfig.context.get("swiftFlags");
+        assertTrue(swiftFlags.contains("arm64-apple-macosx{{osMinVersion}}"));
 
         List<String> flags = (List<String>)platformConfig.context.get("flags");
         assertFalse(flags.contains("-arch"));
         assertFalse(flags.contains("-target"));
         assertFalse(flags.contains("-isysroot"));
         assertFalse(flags.contains("{{env.SYSROOT}}"));
+        List<String> libPaths = (List<String>)platformConfig.context.get("libPaths");
+        assertTrue(libPaths.contains("{{env.PLATFORMSDK_DIR}}/XcodeDefault{{env.XCODE_VERSION}}.xctoolchain/usr/lib/swift/macosx"));
 
         Map<String, String> env = new HashMap<>();
         env.put("SYSROOT", "/opt/platformsdk/MacOSX.sdk");
@@ -451,6 +455,10 @@ public class ExtenderTest {
         assertFalse(legacyRenderedCompileCmd.contains("-target "));
         assertFalse(legacyRenderedCompileCmd.contains("-m64 "));
         assertTrue(legacyRenderedCompileCmd.contains("-isysroot /opt/platformsdk/MacOSX.sdk"));
+
+        PlatformConfig x86_64PlatformConfig = mergePlatformConfig(config, "x86_64-osx");
+        List<String> x86_64SwiftFlags = (List<String>)x86_64PlatformConfig.context.get("swiftFlags");
+        assertTrue(x86_64SwiftFlags.contains("x86_64-apple-macosx{{osMinVersion}}"));
     }
 
     @Test

--- a/server/src/test/java/com/defold/extender/services/cocoapods/PodUtilsTest.java
+++ b/server/src/test/java/com/defold/extender/services/cocoapods/PodUtilsTest.java
@@ -74,7 +74,9 @@ public class PodUtilsTest {
     public void testSwiftModuleName() throws ExtenderException {
         assertEquals("arm64-apple-ios", PodUtils.swiftModuleNameFromPlatform("arm64-ios"));
         assertEquals("x86_64-apple-ios-simulator", PodUtils.swiftModuleNameFromPlatform("x86_64-ios"));
+        assertEquals("arm64-apple-macos", PodUtils.swiftModuleNameFromPlatform("arm64-osx"));
         assertEquals("arm64-apple-macos", PodUtils.swiftModuleNameFromPlatform("arm64-macos"));
+        assertEquals("x86_64-apple-macos", PodUtils.swiftModuleNameFromPlatform("x86_64-osx"));
         assertEquals("x86_64-apple-macos", PodUtils.swiftModuleNameFromPlatform("x86_64-macos"));
         assertThrows(ExtenderException.class, () -> { PodUtils.swiftModuleNameFromPlatform("x86_64-linux"); });
 

--- a/server/test-data/sdk/a/defoldsdk/extender/build.yml
+++ b/server/test-data/sdk/a/defoldsdk/extender/build.yml
@@ -135,7 +135,7 @@ platforms:
     context:
         frameworks:     ["Foundation", "AppKit", "Cocoa", "OpenGL", "OpenAL", "AGL", "IOKit", "Carbon", "CoreVideo", "QuartzCore"]
         engineLibs:     ["engine", "engine_service", "mbedtls", "zip", "profile", "remotery", "profilerext", "record", "gameobject", "ddf", "resource", "gamesys", "gamesys_model", "gamesys_rig", "script_box2d", "graphics", "graphics_transcoder_basisu", "basis_transcoder", "physics", "BulletDynamics", "BulletCollision", "platform", "LinearMath", "Box2D", "render", "render_font_default", "script", "luajit-5.1", "extension", "hid", "input", "particle", "rig", "dlib", "image", "glfw3", "gui", "crashext", "sound", "tremolo", "vpx", "liveupdate"]
-        libPaths:       ["{{env.SYSROOT}}/usr/lib/swift", "{{env.PLATFORMSDK_DIR}}/XcodeDefault{{env.XCODE_VERSION}}.xctoolchain/usr/lib/swift-{{env.SWIFT_VERSION}}/macosx", "{{env.PLATFORMSDK_DIR}}/XcodeDefault{{env.XCODE_VERSION}}.xctoolchain/usr/lib/clang/{{env.XCODE_CLANG_VERSION}}/lib/darwin"]
+        libPaths:       ["{{env.SYSROOT}}/usr/lib/swift", "{{env.PLATFORMSDK_DIR}}/XcodeDefault{{env.XCODE_VERSION}}.xctoolchain/usr/lib/swift-{{env.SWIFT_VERSION}}/macosx", "{{env.PLATFORMSDK_DIR}}/XcodeDefault{{env.XCODE_VERSION}}.xctoolchain/usr/lib/swift/macosx", "{{env.PLATFORMSDK_DIR}}/XcodeDefault{{env.XCODE_VERSION}}.xctoolchain/usr/lib/clang/{{env.XCODE_CLANG_VERSION}}/lib/darwin"]
         defines:        ["DM_PLATFORM_MACOS", "DM_PLATFORM_OSX", "GL_DO_NOT_WARN_IF_MULTI_GL_VERSION_HEADERS_INCLUDED"]
         systemIncludes: ["{{env.SYSROOT}}/usr/include/c++/v1"]
         flags:          ["-fno-exceptions", "-fvisibility=hidden", "-Werror=format"]
@@ -181,7 +181,7 @@ platforms:
         linkFlags:      ["-arch", "arm64", "-target", "arm64-apple-darwin19", "-m64"]
         cxxLinkShFlags: ["-arch", "arm64", "-target", "arm64-apple-darwin19", "-m64"]
         zig-target:     'aarch64-macos'
-        swiftFlags:     ["-target", "arm-apple-darwin19"]
+        swiftFlags:     ["-target", "arm64-apple-macosx{{osMinVersion}}"]
 
   x86_64-osx:
     context:
@@ -191,7 +191,7 @@ platforms:
         linkFlags:      ["-arch", "x86_64", "-target", "x86_64-apple-darwin19", "-m64"]
         cxxLinkShFlags: ["-arch", "x86_64", "-target", "x86_64-apple-darwin19", "-m64"]
         zig-target:     'x86_64-macos'
-        swiftFlags:     ["-target", "x86_64-apple-darwin19"]
+        swiftFlags:     ["-target", "x86_64-apple-macosx{{osMinVersion}}"]
 
   ios:
     env:

--- a/server/test-data/sdk/a/defoldsdk/extender/build.yml
+++ b/server/test-data/sdk/a/defoldsdk/extender/build.yml
@@ -138,7 +138,7 @@ platforms:
         libPaths:       ["{{env.SYSROOT}}/usr/lib/swift", "{{env.PLATFORMSDK_DIR}}/XcodeDefault{{env.XCODE_VERSION}}.xctoolchain/usr/lib/swift-{{env.SWIFT_VERSION}}/macosx", "{{env.PLATFORMSDK_DIR}}/XcodeDefault{{env.XCODE_VERSION}}.xctoolchain/usr/lib/clang/{{env.XCODE_CLANG_VERSION}}/lib/darwin"]
         defines:        ["DM_PLATFORM_MACOS", "DM_PLATFORM_OSX", "GL_DO_NOT_WARN_IF_MULTI_GL_VERSION_HEADERS_INCLUDED"]
         systemIncludes: ["{{env.SYSROOT}}/usr/include/c++/v1"]
-        flags:          ["-O2", "-g", "-stdlib=libc++", "-mmacosx-version-min={{osMinVersion}}", "-isysroot", "{{env.SYSROOT}}", "-nostdinc++", "-fno-exceptions", "-fvisibility=hidden", "-Werror=format"]
+        flags:          ["-fno-exceptions", "-fvisibility=hidden", "-Werror=format"]
         linkFlags:      ["-O2", "-g", "-stdlib=libc++", "-mmacosx-version-min={{osMinVersion}}", "-isysroot", "{{env.SYSROOT}}", "-Wl,-rpath,@loader_path/.", "-Wl,-U,_objc_loadClassref"]
         cxxLinkShFlags: ["-O2", "-g", "-stdlib=libc++", "-mmacosx-version-min={{osMinVersion}}", "-isysroot", "{{env.SYSROOT}}", "-Wl,-rpath,@loader_path/.", "-Wl,-U,_objc_loadClassref"]
         libs:           ['clang_rt.osx']
@@ -162,7 +162,7 @@ platforms:
     zipContentPattern: 'dmengine'
     allowedLibs:    ["(\\w[\\w\\.+-]+)"]
     allowedFlags: ["-ObjC","-ObjC\\+\\+","-Wa,{{comma_separated_arg}}","-W{{warning}}","-Wno-{{warning}}","-ansi","--ansi","-std-default={{arg}}","-stdlib=(libstdc\\+\\+|libc\\+\\+)","-w","-std=(c89|c99|c\\+\\+98|c\\+\\+0x|c\\+\\+11|c\\+\\+14|c\\+\\+17|c\\+\\+20)","-Wp,{{comma_separated_arg}}","--extra-warnings","--warn-{{warning}}","--warn-={{warning}}","-f(no-)?objc-arc(-exceptions)?","-ferror-limit={{number}}","-O([0-4]?|fast|s|z)","-Wl,[-_a-zA-Z0-9,/@]+","-fno-rtti","-f\\w[\\w-=]+","-x","c"]
-    compileCmd: 'clang++ -c {{#systemIncludes}}-isystem {{{.}}} {{/systemIncludes}} {{#defines}}-D{{{.}}} {{/defines}} {{#flags}}{{{.}}} {{/flags}} {{#ext.includes}}-I{{{.}}} {{/ext.includes}} {{#ext.frameworkPaths}}-F{{{.}}} {{/ext.frameworkPaths}} {{#includes}}-I{{{.}}} {{/includes}} {{src}} -o{{tgt}}'
+    compileCmd: 'clang++ -c {{#clangArch}}-arch {{{clangArch}}} {{/clangArch}}{{#clangTarget}}-target {{{clangTarget}}} {{/clangTarget}}{{#clangArch}}-m64 {{/clangArch}}-O2 -g -stdlib=libc++ -mmacosx-version-min={{osMinVersion}} -isysroot {{env.SYSROOT}} -nostdinc++ {{#systemIncludes}}-isystem {{{.}}} {{/systemIncludes}} {{#defines}}-D{{{.}}} {{/defines}} {{#flags}}{{{.}}} {{/flags}} {{#ext.includes}}-I{{{.}}} {{/ext.includes}} {{#ext.frameworkPaths}}-F{{{.}}} {{/ext.frameworkPaths}} {{#includes}}-I{{{.}}} {{/includes}} {{src}} -o{{tgt}}'
     linkCmd: 'clang++ -o {{tgt}} {{#linkFlags}}{{{.}}} {{/linkFlags}} {{#ext.libPaths}}-L{{{.}}} {{/ext.libPaths}} {{#libPaths}}-L{{{.}}} {{/libPaths}} {{#libs}}-l{{{.}}} {{/libs}} {{#ext.libs}}-l{{{.}}} {{/ext.libs}} {{#engineLibs}}-l{{{.}}} {{/engineLibs}} {{#ext.dynamicLibs}}-l{{{.}}} {{/ext.dynamicLibs}} {{#dynamicLibs}}-l{{{.}}} {{/dynamicLibs}} {{#ext.frameworkPaths}}-F{{{.}}} {{/ext.frameworkPaths}} {{#ext.frameworks}}-framework {{{.}}} {{/ext.frameworks}} {{#frameworks}}-framework {{{.}}} {{/frameworks}} {{#src}}{{{.}}} {{/src}}'
     libCmd: 'ar rcs {{tgt}} {{#objs}}{{{.}}} {{/objs}}'
     symbolCmd: 'dsymutil -o {{src}}.dSYM {{src}}'
@@ -176,7 +176,8 @@ platforms:
   arm64-osx:
     context:
         libPaths:       ["{{dynamo_home}}/lib/arm64-macos", "{{dynamo_home}}/ext/lib/arm64-macos"]
-        flags:          ["-arch", "arm64", "-target", "arm64-apple-darwin19", "-m64"]
+        clangArch:      "arm64"
+        clangTarget:    "arm64-apple-darwin19"
         linkFlags:      ["-arch", "arm64", "-target", "arm64-apple-darwin19", "-m64"]
         cxxLinkShFlags: ["-arch", "arm64", "-target", "arm64-apple-darwin19", "-m64"]
         zig-target:     'aarch64-macos'
@@ -185,7 +186,8 @@ platforms:
   x86_64-osx:
     context:
         libPaths:       ["{{dynamo_home}}/lib/x86_64-macos", "{{dynamo_home}}/ext/lib/x86_64-macos"]
-        flags:          ["-arch", "x86_64", "-target", "x86_64-apple-darwin19", "-m64"]
+        clangArch:      "x86_64"
+        clangTarget:    "x86_64-apple-darwin19"
         linkFlags:      ["-arch", "x86_64", "-target", "x86_64-apple-darwin19", "-m64"]
         cxxLinkShFlags: ["-arch", "x86_64", "-target", "x86_64-apple-darwin19", "-m64"]
         zig-target:     'x86_64-macos'

